### PR TITLE
Add scio-algebra

### PR DESCRIPTION
**Pull request by regadas**
The main purpose of this new module is to ease not only provide a more natural way to interact and manipulate more "complex" types but also to ease up the usage of `cats` in `scio`.

### CODE REVIEW

**Comment by jto** [comment]
@regadas is this still a WIP ?

**Comment by regadas** [comment]
Hi @jto! yeah I'll wrap this up soon™ and open it.

File: scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+final class HigherKindCatsOps[F[_], A](private val coll: SCollection[F[A]]) extends AnyVal {
+
+  /**
+   * Replaces the `A` value in `F[A]` with the supplied value.
+   *
+   * Example:
+   *
+   * {{{
+   * scala> val coll: SCollection[List[Int]] = sc.paralelize(Seq(List(1, 2, 3)))
+   * scala> coll.as("hello") // List(hello, hello, hello)
```
Comment (anish749) [inline scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala]: Let's write this as : `val out: SCollection[List[Int]] = coll.as("hello") // SCollection(List(hello, hello, hello))`

Comment (anish749) [reply scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala]: I am not sure why I got confused with this.

File: scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+   */
+  def fold_(implicit M: Monoid[A], C: Coder[A]): SCollection[A] =
+    coll.context.wrap {
+      val combine = Combine.globally(new ReduceFn[A] {
+        override val vacoder = Coder[JList[A]]
+        override val vocoder = C
+
+        override def reduceOption(accumulator: JIterable[A]): Option[A] =
+          Fns
+            .reduceOption(accumulator) { (a, b) =>
```
[the remark this answers is not in the document]
Comment (regadas) [reply scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala]: Hi @Daenyth very late reply; yup good point!

**Pull request closed without merging**